app/models/mixins/relationship_mixin.rb

```diff
@@ -580,9 +580,13 @@ def child_types(*args)
   end
 
   def add_parent(parent)
-    return update(:parent => parent) if use_ancestry?
-
-    parent.with_relationship_type(relationship_type) { parent.add_child(self) }
+    parent.with_relationship_type(relationship_type) do
+      if use_ancestry?
+        update(:parent => parent)
+      else
+        parent.add_child(self)
+      end
+    end
```

Comment:
do we need this change anymore?

you already handle `parent.with_relationship_type` in `parent=`